test/cpp/jit/test_gpu.cpp

```diff
@@ -13381,6 +13381,142 @@ TEST(NVFuserTest, FusionValidateParallelize5_CUDA) {
   fe.compileFusion(&fusion);
 }
 
+TEST(NVFuserTest, FusionDAGMerging_CUDA) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  auto tv0 = makeSymbolicTensor(5);
+  auto tv1 = makeSymbolicTensor(1);
+  fusion.addInput(tv0);
+  fusion.addInput(tv1);
+
+  // Branch 0
+  auto tv2 = sum(tv0, {0}); // 0
+  auto tv3 = sum(tv2, {0}); // 1
+  auto tv4 = sum(tv3, {0}); // 2
+  auto tv5 = sum(tv4, {0}); // 3
+
+  // Branch 1
+  auto tv6 = add(tv1, new Double(1)); // 4
+
+  // Merge
+  auto tv7 = add(tv6, tv5); // 5
+
+  // Maximum expected output groups (can improve overtime):
+  //  {0}, {1}, {2}, {3,4,5}
+  //  without final merge would have been {0}, {1}, {2}, {3,4}, {5}
```

Comment:
I'm confused with this comment. Does each number mean a tensor? E.g., `{0}` means `{tv0}`? If so, what about `tv6` and `tv7`?

Reply:
The expression number, which is in the order of `fusion::exprs()`